cmd/soroban-cli/src/commands/tx/edit.rs

```diff
@@ -0,0 +1,184 @@
+      "fee": 100,
+      "seq_num": 0,
+      "cond": "none",
+      "memo": "none",
+      "operations": [],
+      "ext": "v0"
+    },
+    "signatures": []
+  }
+}
+"#;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    StellarXdr(#[from] stellar_xdr::curr::Error),
+
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::Error),
+
+    #[error(transparent)]
+    Base64Decode(#[from] base64::DecodeError),
+
+    #[error("Editor returned non-zero status")]
+    EditorNonZeroStatus,
+}
+
+// Command to edit the transaction
+/// e.g. `stellar tx new manage-data --data-name hello --build-only | stellar tx edit`
+#[derive(Debug, clap::Parser, Clone, Default)]
+#[group(skip)]
+pub struct Cmd {}
+
+impl Cmd {
+    pub fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let print = Print::new(global_args.quiet);
+        let json: String = if stdin().is_terminal() {
+            DEFAULT_JSON.to_string()
+        } else {
+            let mut input = String::new();
+            stdin().read_line(&mut input)?;
+            let input = input.trim();
+            xdr_to_json::<curr::TransactionEnvelope>(input)?
+        };
+
+        let path = tmp_file(&json)?;
+        let editor = get_editor();
+
+        open_editor(&print, &editor, &path)?;
+
+        let contents = fs::read_to_string(&path)?;
+        let xdr = json_to_xdr::<curr::TransactionEnvelope>(&contents)?;
+        fs::remove_file(&path)?;
+
+        println!("{xdr}");
+
+        Ok(())
+    }
+}
+
+struct Editor {
+    cmd: String,
+    source: String,
+    args: Vec<String>,
+}
+
+fn tmp_file(contents: &str) -> Result<PathBuf, Error> {
+    let temp_dir = env::current_dir().unwrap_or(env::temp_dir());
+    let file_name = format!("stellar-json-xdr-{}.json", rand::random::<u64>());
+    let path = temp_dir.join(file_name);
+
+    let mut file = fs::File::create(&path)?;
+    file.write_all(contents.as_bytes())?;
+
+    Ok(path)
+}
+
+fn get_editor() -> Editor {
+    let (source, cmd) = env::var("STELLAR_EDITOR")
+        .map(|val| ("STELLAR_EDITOR", val))
+        .or_else(|_| env::var("EDITOR").map(|val| ("EDITOR", val)))
+        .or_else(|_| env::var("VISUAL").map(|val| ("VISUAL", val)))
+        .unwrap_or_else(|_| ("default", "vim".to_string()));
+
+    let parts: Vec<&str> = cmd.split_whitespace().collect();
+    let cmd = parts[0].to_string();
+    let args = &parts[1..]
+        .iter()
+        .map(|&s| s.to_string())
+        .collect::<Vec<String>>();
+
+    Editor {
+        source: source.to_string(),
+        cmd,
+        args: args.clone(),
+    }
+}
+
+fn open_editor(print: &Print, editor: &Editor, path: &PathBuf) -> Result<(), Error> {
+    print.infoln(format!(
+        "Using `{source}=\"{cmd}\"`",
+        source = editor.source,
+        cmd = editor.cmd,
+    ));
+
+    print.infoln("Opening editor...".to_string());
```

Comment:
I think these print lines could be merged:
```suggestion
    print.infoln(format!(
        "Opening editor with `{source}=\"{cmd}\"`",
        source = editor.source,
        cmd = editor.cmd,
    ));
```